packages/roosterjs-editor-api/lib/utils/processList.ts

```diff
@@ -5,76 +5,78 @@ import {
     isVoidHtmlElement,
     isBlockElement,
     Browser,
-    isNodeEmpty,
-    getTagOfNode,
-    getComputedStyles,
-    applyFormat,
-    getFirstLeafNode,
+    getSelectionPath,
+    getRangeFromSelectionPath,
 } from 'roosterjs-editor-dom';
 
 const TEMP_NODE_CLASS = 'ROOSTERJS_TEMP_NODE_FOR_LIST';
-const TEMP_NODE_HTML = `<img class="${TEMP_NODE_CLASS}">`;
+const TEMP_NODE_HTML = "<img class=\"" + TEMP_NODE_CLASS + "\">";
+//const SELECTION_NODE_CLASS = 'ROOSTERJS_SELECTION_NODE';
+//const SELECTION_NODE_HTML = "<img class=\"" + SELECTION_NODE_CLASS + "\">";
+
+type ValidProcessListDocumentCommands = DocumentCommand.Outdent | DocumentCommand.Indent | DocumentCommand.InsertOrderedList | DocumentCommand.InsertUnorderedList;
 
 /**
  * Browsers don't handle bullet/numbering list well, especially the formats when switching list statue
  * So we workaround it by always adding format to list element
  */
-export default function processList(editor: Editor, command: DocumentCommand): Node {
-    if (Browser.isChrome) {
+export default function processList(editor: Editor, command: ValidProcessListDocumentCommands): Node {
+    let clonedNode: Node;
+    let relativeSelectionPath;
+    if (Browser.isChrome && command == DocumentCommand.Outdent) {
+        const parentLINode =  editor.getElementAtCursor('LI');
+        if (parentLINode) {
+            let currentRange = editor.getSelectionRange();
+            relativeSelectionPath = getSelectionPath(parentLINode, currentRange);
+            // Chrome has some bad behavior when outdenting
+            // in order to work around this, we need to take steps to deep clone the current node
+            // after the outdent, we'll replace the new LI with the cloned content.
+            clonedNode =  parentLINode.cloneNode(true);
+        }
+
         workaroundForChrome(editor);
     }
 
-    let currentNode = editor.getElementAtCursor();
-    let currentFormat = getComputedStyles(currentNode);
     let existingList = editor.getElementAtCursor('OL,UL');
     editor.getDocument().execCommand(command, false, null);
-    editor.queryElements('.' + TEMP_NODE_CLASS, node => editor.deleteNode(node));
+    let newParentNode: HTMLElement;
+    editor.queryElements('.' + TEMP_NODE_CLASS, node => {
+        newParentNode = node.parentElement;

```

Comment:
Suggest use parentNode.

Reply:
Perticular reason for this? Or just naming.

Reply:
In IE11, parentElement only works when current node is not Text node. Here it should be ok, but I'd like to always use parentNode for safety.

Reply:
I see. Since this code will only run for chrome I'd prefer keeping it here to prevent messy typecasting situations, and to be more confident in the behavior we'd be executing.